auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,405 @@
+import re
+try:
+    import pysftp
+except ImportError:
+    raise ImportError(
+        'This module needs pysftp to automaticly write backups to the FTP '
+        'through SFTP.Please install pysftp on your system.'
+        '(sudo pip install pysftp)'
+    )
+from openerp import models, fields, api, _
+from openerp.exceptions import except_orm, Warning
+from openerp import tools
+from openerp import netsvc
+import logging
+_logger = logging.getLogger(__name__)
+
+
+def execute(connector, method, *args):
+    res = False
+    try:
+        res = getattr(connector, method)(*args)
+    except socket.error as e:
+        raise e
+    return res
+
+
+class db_backup(models.Model):
+    _name = 'db.backup'
+
+    def get_connection(self, host, port):
+        uri = 'http://%s:%s' % (host, port)
+        return xmlrpclib.ServerProxy(uri + '/xmlrpc/db')
+
+    def get_db_list(self, host, port):
+        conn = self.get_connection(host, port)
+        db_list = execute(conn, 'list')
+        return db_list
+
+    @api.model
+    def _get_db_name(self):
+        return self.env.cr.dbname
+
+    # Columns local server
+    host = fields.Char(
+        string='Host', default='localhost', size=100, required=True)
+    port = fields.Char(
+        string='Port', default='8069', size=10, required=True)
+    name = fields.Char(
+        string='Database', size=100, required=True,
+        default=_get_db_name,
+        help='Database you want to schedule backups for'
+    )
+    bkp_dir = fields.Char(
+        string='Backup Directory', size=100,
+        default='/odoo/backups',
+        help='Absolute path for storing the backups',
+        required=True
+    )
+    autoremove = fields.Boolean(
+        string='Auto. Remove Backups',
+        help=(
+            "If you check this option you can choose to "
+            "automaticly remove the backup after xx days"
+        )
+    )
+    daystokeep = fields.Integer(
+        string='Remove after x days',
+        default=30,
+        help=(
+            "Choose after how many days the backup should be "
+            "deleted. For example:\nIf you fill in 5 the backups "
+            "will be removed after 5 days."
+        ), required=True
+    )
+    sftpwrite = fields.Boolean(
+        string='Write to external server with sftp',
+        help=(
+            "If you check this option you can specify the details "
+            "needed to write to a remote server with SFTP."
+        )
+    )
+    sftppath = fields.Char(
+        string='Path external server',
+        help=(
+            "The location to the folder where the dumps should be "
+            "written to. For example /odoo/backups/.\nFiles will then"
+            " be written to /odoo/backups/ on your remote server."
+        )
+    )
+    sftpip = fields.Char(
+        string='IP Address SFTP Server',
+        help=(
+            "The IP address from your remote"
+            " server. For example 192.168.0.1"
+        )
+    )
```

Comment:
well everything points to "IP address" not to "DNS name" so, stumbling on this while reading the code I'm wondering if there is something which prevents using a FQDN.